### ISSUE TITLE
Software Portrait view for Samsung DEX

### ISSUE THREAD

**Issue by Toldwin** [tachiyomiorg/tachiyomi#10247]
### Describe your suggested feature

Hi,

I just bought a tactile screen to use it as a tablet by docking my phone. When I do this Samsung DEX activate and I can use my phone as a tablet but it's locked in landscape mode and DEX is a special UI where screen rotation can't change neither physicaly or with Android API. So when I use the android application any portrait or landscape option available and working neither in Tachiyomi or any other app.

(It has been told that after years of user reclamations, Samsung DEX that was shipped with ONE UI 6.0 can now use portrait mode. It's true but only working if you own a tablet : The functionnality is not implemented on Android phones...)

So I would like to know if there would be a way to activate a software Portrait view where the Android screen rotation API would not be used and where in landscape mode you could see the page like in portrait mode. This would be so good for people like me using in an environnement where Android screen rotation is not available.

Due to the few user impacted my request will probably not be implemented but anyway I thank you a lot for this great apps !
Regards

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.14.7](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

**Comment by arkon** [comment]
I don't understand what this is requesting. Is the UI rotated the wrong way or something?

**Comment by Toldwin** [comment]
In Tachiyomi, on my phone, standard UI, all work fine.

But when I use the rotation options in Tachiyomi Samsung Dex (By default, free, portrait, landscape, blocked on portrait, blocked on landscape) nothing change at all. My Samsung DEX view on my external monitor stay in landscape mode (with a 16/9 view). 

Samsung DEX open android apps in a special window to allow 16/9 rendering. I think this window is not allowing all Android API functions and specially the rotation one.

![image](https://github.com/tachiyomiorg/tachiyomi/assets/12753133/c3686426-88d7-440d-bf0d-c945016867dc)

**Comment by arkon** [comment]
See you saying DEX is landscape, but your monitor is portrait and you want the app to render in a fake rotated mode to compensate?

**Comment by Toldwin** [comment]
Sorry, hard to explain this clearly  :)

The best is perhaps to make a draft... Does it help ? 

![image](https://github.com/tachiyomiorg/tachiyomi/assets/12753133/2c68c6c4-e0b2-4393-9295-c2fd21d6b80e)

**Comment by Toldwin** [comment]
> So you're saying DEX is landscape, but your monitor is portrait and you want the app to render in a fake rotated mode to compensate?

Yes exactly. Render a software (not linked with Android) rotated mode to compensate the fact that DEX is blocked.
I think tought that current DEX rending mode is in fact a Portrait mode (for android) but stretched in 16/9. 

So it should be a Portrait mode (even if rendered by DEX as Landescape) with the pages going 90 degree slided to the right and with the 1 on the right followed by the 2 (...) on the left.

In fact I can use my screen in Landscape or in Portrait but I would like to use it in Portrait to have a way better usage of the screen

**Comment by arkon** [comment]
Sorry I'm going to turn this down since this seems very specific to your pretty janky setup. You could consider rotating your monitor.

**Comment by mystic-man-rhino** [comment]
Try changing reading modes in the app.

**Comment by Toldwin** [comment]
Thanks mystic-man-rhino for the reply.

I did tested all option and I can change the reading style "left yo right swap" or "up to bottom page"...

But no option is working to see page in full screen.

When in DEX Landscape (=Android portrait mode) the page is Ok if seen as standard monitor. 
But if I use my screen in Portrait instead the Android portrait mode say the same and my page is wrongly orientated.

Any Portrait or Landscape (forced or not) has any impact on my view. Samsung DEX API seems to nullify screen orientation API call.

I will try to see if I can make a MR about this. Tachiyomi is open source and that's awesome.
I checked fast and that doesn't seem simple. I will need a lot of time to understand the app behaviours.

**Comment by Toldwin** [comment]
I just did an ugly but working solution : 
![image](https://github.com/tachiyomiorg/tachiyomi/assets/12753133/b95f1d57-9847-4a6c-818a-59aff99a84f8)

My issue to do a good implementation is that it should be made in the Tile "SubsamplingScaleImageView" from com.davemorrissey.labs.subscaleview...

You did a great job with the project, open and building it was flowless even with no special Android background.
